tests/integration/nodes/test_memory.py

```diff
@@ -0,0 +1,81 @@
+import os
+import uuid
+
+import pytest
+
+from dynamiq import Workflow, connections, flows, prompts
+from dynamiq.memory import Config, Memory
+from dynamiq.memory.backend import InMemory
+from dynamiq.nodes.agents.simple import SimpleAgent
+from dynamiq.nodes.llms import OpenAI
+from dynamiq.prompts import MessageRole
+from dynamiq.runnables import RunnableStatus
+
+# Constants
+AGENT_ROLE = "helpful assistant"
+AGENT_GOAL = "is to provide useful information and answer questions"
+
+
+@pytest.fixture
+def openai_connection():
+    return connections.OpenAI(
+        id=str(uuid.uuid4()),
+        api_key=os.getenv("OPENAI_API_KEY"),
+    )
```

Comment:
I think if we use tests with real credentials, we should move this test into a separate folder [integration_with_creds](https://github.com/dynamiq-ai/dynamiq/tree/main/tests/integration_with_creds)